src/Service.php

```diff
@@ -87,5 +87,31 @@ public function publishSupervisorNoop() {
 		}
 	}
 
+	/**
+	 * Metoda, která smaže všechny záznamy z DB s nastaveným stavem STATE_DONE.
+	 *
+	 * @param array $callbacksNames nepovinný parametr pro výběr konkrétních callbacků
+	 */
+	public function clearDoneRecords($callbacksNames = []) {
+
+		$ago = (new \Nette\Utils\DateTime('midnight'))->modify("-".$this->config["deleteOldBefore"]);
+
+		$state = Entity\QueueEntity::STATE_DONE;
+		$sql = "DELETE FROM rabbit_queue WHERE created <= '$ago' AND state = '$state'";
+
+		if (!empty($callbacksNames)) {
+			$sql .= "AND callbackName IN (";
```

Comment:
Tady chybí před AND mezera, když se to konkatenuje. Ale stejně by bylo lepší použít QB.